filler.go

```diff
@@ -0,0 +1,126 @@
+package rocket
+
+import (
+	"encoding/json"
+	"io"
+	"net/url"
+	"reflect"
+)
+
+type (
+	contextFiller interface {
+		fill(context reflect.Value) error
+	}
+	nextFiller struct {
+		next contextFiller
+	}
+	routeFiller struct {
+		nextFiller
+		h  *handler
+		rs []string
+	}
+	queryFiller struct {
+		nextFiller
+		h     *handler
+		query url.Values
+	}
+	jsonFiller struct {
+		nextFiller
+		h    *handler
+		body io.Reader
+	}
+	formFiller struct {
+		nextFiller
+		h    *handler
+		form url.Values
+	}
+)
+
+func defaultFillerChain(h *handler, rs []string, body io.Reader, query, form url.Values) contextFiller {
+	return newRouteFiller(h, rs,
+		newQueryFiller(h, query,
+			newJSONFiller(h, body,
+				newFormFiller(h, form, nil),
+			),
+		),
+	)
+}
```

Comment:
I prefer this way
```
type Chain struct {
  context reflect.Value
  error errors
}

func (c *Chain) pipe (f *filler) {
  if c.error == nil {
    c.error = f.fill(c.context);
  }
}

err := newChain(context).
  pipe(newRouteFiller(h, routes)).
  pipe(newQueryFiller(h, query)).
  pipe(newJSONFiller(h, body)).
  pipe(newFormFiller(h, form)).
  error
```

Reply:
Has a problem, JSON filler would stop form filler if it matched

Reply:
Or change fill interface, but I think both are over design now.

Reply:
but right now, `defaultFillerChain` need to know every argument of every filler, if we want to create a new filler like `binaryFiller` for example, then we need to change `defaultFillerChain` and its caller, our arguments hell grow also, so I think is better to fix it now.

Reply:
I move default chain into handler, would it make sense about let it know what info should handler know now?